### PR TITLE
Fix iterations.php request for login

### PR DESCRIPTION
**Pull request by brendanlong**
For some reason iterations.php only works with GET requests now and is flaky, so this:

- Fixes the error message when there's no error `cause` field so it actually shows something instead of "unknown"
- Uses a request to login.php to get the number of iterations

Fixes https://github.com/lastpass/lastpass-cli/issues/604

### CODE REVIEW

**Comment by rkirkpat** [comment]
Appears the CI build and test has bit-rotted, failures are due to apt and brew errors it appears. 

Was able to apply with brew, build, and verify it fixed the `iterations.php` breakage, see: <https://github.com/lastpass/lastpass-cli/issues/604#issuecomment-843612877>

**Comment by brendanlong** [comment]
I updated this to use login.php since the previous version using iterations.php didn't work for everyone.

File: endpoints.c [hunk 1]
```diff
@@ -48,14 +48,20 @@ unsigned int lastpass_iterations(const char *username)
 {
 	_cleanup_free_ char *reply = NULL;
 	_cleanup_free_ char *user_lower = NULL;
+	_cleanup_free_ char *path = NULL;
```
Comment (scooter-dangle) [inline endpoints.c:51]: Where is `path` used?

```suggestion
```

**Comment by crispjw** [comment]
I recently updated my Password Iterations to 310,000 per OWASP recommendations and that broke the lpass command line for me.  When I rebuilt lastpass with this code fix, the lpass command line works again.  Thanks!

**Comment by 0xdevalias** [comment]
@mateusmartins-lp It's good 'open source courtesy' to give context when closing an issue/PR, not just closing it with no explanation.

**Comment by 0xdevalias** [comment]
> > @mateusmartins-lp It's good 'open source courtesy' to give context when closing an issue/PR, not just closing it with no explanation.
> 
> Hello @0xdevalias. The Pull Requests were closed unintentionally. By tomorrow I will be sure to leave a message on how to proceed. My sincere apologies for the inconvenience.
> 
> _Originally posted by @mateusmartins-lp in https://github.com/lastpass/lastpass-cli/issues/553#issuecomment-2098844351_

**Comment by mateusmartins-lp** [comment]
Regrettably, your PR submission was unintentionally closed during an operation, before we could complete our review and respond accordingly. Unable to revert it to 'Open' status, we invite you to resubmit your contribution at your earliest convenience. We apologize for this mishap. Rest assured, we value all contributions and remain dedicated to providing transparency and closure to the community. Thank you for your understanding.

**Pull request closed without merging**